### PR TITLE
#68 docker dev env

### PR DESCRIPTION
**Pull request by LuvvAggarwal**
Hey @juliaferraioli,  I have created a .devcontainer directory with Dockerfile. But I am facing some issues in running Hugo server I think it is related to the root path.
Just FYI, I have also tried the Dev Env(beta) feature of Docker which created a compose.yaml file automatically.
SS of compose-dev.yaml
image

Any help/suggestion would be really helpful, Thanks

### CODE REVIEW

**Comment by juliaferraioli** [comment]
Heya, this is looking great so far. Is there some reason you closed the PR?

If you'd like to keep working on it, can you share the error message you're experiencing?

**Comment by LuvvAggarwal** [comment]
Below is the error that I am facing after running the container
image

**Comment by juliaferraioli** [comment]
Weird. It works for me. Can you run an `ls` and show the output? This is what I see when I've checked out your PR.

image

**Comment by LuvvAggarwal** [comment]
Is the latest commit fine?

File: archetypes/stories.md
Comment (juliaferraioli) [inline archetypes/stories.md:1]: This file shouldn't be in this commit.

File: content/stories/2023/thea-flowers.md
Comment (juliaferraioli) [inline content/stories/2023/thea-flowers.md:1]: This file shouldn't be in this commit.

File: layouts/partials/credits.html
Comment (juliaferraioli) [inline layouts/partials/credits.html:1]: This file shouldn't be in this commit.

File: layouts/stories/single.html
Comment (juliaferraioli) [inline layouts/stories/single.html:1]: This file shouldn't be in this commit.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    export ARCH=64bit ;; \
+    esac && \
+    echo ${ARCH} && \
+    wget -O ${VERSION}.tar.gz https://github.com/gohugoio/hugo/releases/download/v${VERSION}/${VARIANT}_${VERSION}_Linux-${ARCH}.tar.gz && \
+    tar xf ${VERSION}.tar.gz && \
+    mv hugo /usr/bin/hugo
+
+# Hugo dev server port
+EXPOSE 1313
+CMD ["hugo" , "server", "-D", "--bind", "0.0.0.0" ]
```
Comment (juliaferraioli) [inline Dockerfile:38]: Can you explain why you included this line? I'd like to keep it as minimal as possible.

**Review by juliaferraioli — CHANGES_REQUESTED** [review]
I'm a bit confused by these changes. The two files that we need should be in a `.devcontainer` directory:

- Dockerfile
- devcontainer.json

You can see the template to work off of in the [microsoft repository](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/hugo/.devcontainer/devcontainer.json). Let's keep this PR limited to that scope.

Please ensure that you only commit the files that you are actively changing.

File: Dockerfile (hunk 1 above)
Comment (LuvvAggarwal) [reply Dockerfile]: I was facing some errors so tried binding
Also, I understand my previous approach was not correct
Hope this commit is correct

**Review by juliaferraioli — CHANGES_REQUESTED** [review]
Okay, this looks good! One final request -- remove everything but the `.devcontainer` folder and contents, the `.gitignore` file, and `DEVELOPMENT.md` file from the PR. After that I will approve and merge!

**Comment by LuvvAggarwal** [comment]
> Okay, this looks good! One final request -- remove everything but the `.devcontainer` folder and contents, the `.gitignore` file, and `DEVELOPMENT.md` file from the PR. After that I will approve and merge!

Can I close the PR and create a new Branch for this? I am unaware of how can I remove selected files from the commit
Please guide

**Comment by juliaferraioli** [comment]
> > Okay, this looks good! One final request -- remove everything but the `.devcontainer` folder and contents, the `.gitignore` file, and `DEVELOPMENT.md` file from the PR. After that I will approve and merge!
> 
> Can I close the PR and create a new Branch for this? I am unaware of how can I remove selected files from the commit Please guide

I wouldn't close the PR! Whenever you accidentally commit files you didn't mean to, you can follow these steps:

1. `git checkout HEAD^ -- <file>` (replace <file> with each unwanted file)
1. `git commit -m "remove unnecessary files from commit" (commit the removals)
1. `git push` (push to your branch

You can read more about what this does and more details on [this SO post](https://stackoverflow.com/questions/18357511/git-remove-committed-file-after-push)

Let me know if that doesn't work the way I think it should or if it doesn't make sense.

**Comment by juliaferraioli** [comment]
Okay, I appreciate your bearing with me on this! I've done some git wizardry that I admit I don't fully understand but will try to so I can explain better next time.

That said, in the future, make sure that your main branch in your fork is synced with upstream before checking out a new branch to make changes. It'll make contributing _much_ easier.

@all-contributors please add @LuvvAggarwal for infra improvements!

**Pull request merged**